ddtrace/contrib/cassandra/session.py

```diff
@@ -31,32 +36,116 @@ def traced_connect(func, instance, args, kwargs):
     session = func(*args, **kwargs)
     if not isinstance(session.execute, wrapt.FunctionWrapper):
         # FIXME[matt] this should probably be private.
-        setattr(session, 'execute', wrapt.FunctionWrapper(session.execute, traced_execute))
+        setattr(session, 'execute_async', wrapt.FunctionWrapper(session.execute_async, traced_execute_async))
     return session
 
-def traced_execute(func, instance, args, kwargs):
-    cluster = getattr(instance, 'cluster', None)
+def traced_set_final_result(func, instance, args, kwargs):
+    result = kwargs.get('result') or args[0]
+    span = getattr(instance, CURRENT_SPAN, None)
+    if not span:
+        log.debug('traced_set_final_result was not able to get the current span from the ResponseFuture')
+        return func(*args, **kwargs)
+    with span:
+        span.set_tags(_extract_result_metas(cassandra.cluster.ResultSet(instance, result)))
+    return func(*args, **kwargs)
+
+def traced_set_final_exception(func, instance, args, kwargs):
+    exc = kwargs.get('result') or args[0]
+    span = getattr(instance, CURRENT_SPAN, None)
+    if not span:
+        log.debug('traced_set_final_exception was not able to get the current span from the ResponseFuture')
+        return func(*args, **kwargs)
+    with span:
```

Comment:
How do we want to handle that kind of scenario. I don't think that's the right way to do it since the stacktrace be in our code, which will be confusing to the user.